xdsl/irdl/constraints.py

```diff
@@ -179,13 +140,18 @@ def verifies(self, attr: Attribute) -> TypeGuard[AttributeCovT]:
         except VerifyException:
             return False
 
-    def get_variable_extractors(self) -> dict[str, VarExtractor[Attribute]]:
+    def variables(self) -> set[str]:
         """
-        Get a dictionary of constraint variables to extractors for these variables,
-        which provide a method to obtain the value of each constraint variable from
-        the value of the attribute which is verified by this constraint.
+        Returns a set of the variables that can be extracted by this constraint.
```

Comment:
I would also write something like:
`These variables are always expected to be set by the attribute constraint`